usePortal.ts

```diff
@@ -172,8 +172,13 @@ export default function usePortal({
     }
   }, [isServer, handleOutsideMouseClick, handleKeydown, elToMountTo, portal])
 
-  const Portal = useCallback(({ children }: { children: ReactNode }) => {
-    if (portal.current != null) return createPortal(children, portal.current)
+  const Portal = useCallback(({ children, className }: { children: ReactNode, className: string }) => {
+    if (portal.current != null) {
+      if(className && portal.current){
+        portal.current.className = className
+      }
+      return createPortal(children, portal.current)
+    }
```

Comment:
So, I have not tested this yet, but I think we could actually do it this way
```js
  const Portal = useCallback(({ children, ...props }: { children: ReactNode, className: string }) => {
    if (portal.current != null) return createPortal(children, <div ref={portal} {...props} />)
    return null
  }
```

Reply:
Which would allow for all props not just `className`

Reply:
If we can't get it to work this way, I'll take a closer look asap. We also might need to make the `portal` ref without a default value
```
const portal = useRef()
```
and it's possible we might need to add some checks to make sure it's not `null` in some places

Reply:
This PR is not perfect as the className is only set once. It will not be updated or removed if the props should change.

`createPortal` requires a domNode, e.g. `createPortal(children, domNode)`  

The tricky part is updating the `domNode` properties as the props get updated. I would avoid DOM diffiing stuff. e.g. removing the className or some other attribute if the props should change.

For my use-case the `portalRef` is perfect for me. I can mutate the domNode with `useEffect` and update the domNode if props changes.

This PR should probably be giving some more thoughts :)

Reply:
> I would avoid DOM diffiing stuff. e.g. removing the className or some other attribute if the props should change.

Just curious, but why?

Reply:
Seems complicated :) React does it with it's reconciliation algorithm where it track DOM changes and update accordingly.

It could be simplified in this case by removing all the attributes from the domNode first, and then apply the props to the domNode though.


e.g.  
props might start as: `{className="portal", "data-something:" 123 }`

The domNode is updated to `<div class="portal" data-something="123"></div>`

props gets updated to `{className="portal2", id="my-portal"}`

The domNode is cleared where all attributes are removed `<div></div>`
And it get updated to 
 `<div class="portal2" id="my-portal"></div>`